### PR TITLE
add delete a shopcart item API

### PR DESCRIPTION
**Pull request by momo100110**
add delete a shopcart item API and add test cases in test_models.py and test_routes.py 
image
image

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -254,3 +254,43 @@ def update_shopcart_item(shopcart_id, item_id):
...
+    Delete an item
+
+    This endpoint will delete an item based the id specified in the path
+    """
+   
+    app.logger.info(f"Request to delete item with id='{item_id}' in shopcart with id='{shopcart_id}'.")
+     # See if the shopcart exists and abort if it doesn't
+    shopcart = Shopcart.get_by_id(shopcart_id)
+    if not shopcart:
+        abort(
```
Comment (wl2484) [inline service/routes.py:274]: I have no good answer of why line 274 in `routes.py` is still count as a missing line given the test case you wrote `test_delete_item_shopcart_not_found()` 🤔

Comment (wl2484) [reply service/routes.py:274]: However, you may refer to this [commit](https://github.com/CSCI-GA-2820-SU23-001/shopcarts/pull/47/commits/e0f0fc7ab2dafd3086940cd23705b1c17cd9b047) to add a test case specifically for `shopcart=None`, with this line 274 in `routes.py` would no longer be counted as a line without test!

**Review by wl2484 — COMMENTED** [review]
👍🏻👍🏻👍🏻

File: service/routes.py (hunk 1 above)
Comment (momo100110) [reply service/routes.py:274]: Let me try this approach, thanks for your suggestion!

File: tests/test_routes.py [hunk 2]
```diff
@@ -518,3 +518,57 @@ def test_update_shopcart_item_with_none_shopcart(self):
...
+        resp = self.client.get(
+            f"{BASE_URL}/{shopcart.id}/items/{item_id}",
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_item_not_found(self):
+        """It should not Delete an Item that is not found"""
+        shopcart = self._create_an_empty_shopcart(1)[0]
+        response = self.client.get(f"{BASE_URL}/{shopcart.id}/items/0")
```
Comment (wl2484) [inline tests/test_routes.py:553]: should here be `self.client.delete(...)`?

File: tests/test_routes.py [hunk 3]
```diff
@@ -518,3 +518,57 @@ def test_update_shopcart_item_with_none_shopcart(self):
...
+        res = self.client.post(
+            f"{BASE_URL}/{shopcart.id}/items",
+            json=item.serialize(),
+            content_type="application/json",
+        )
+        data = res.get_json()
+        logging.debug(data)
+
+        with patch('service.models.Shopcart.get_by_id', return_value=None):
+            res = self.client.post(
```
Comment (wl2484) [inline tests/test_routes.py:569]: should here be `self.client.delete(...)`? `/{item_id}` is missing from the API path, either 🙂

**Review by wl2484 — CHANGES_REQUESTED** [review]

File: tests/test_routes.py (hunk 2 above)
Comment (momo100110) [reply tests/test_routes.py:553]: yes, my bad

File: tests/test_routes.py (hunk 3 above)
Comment (momo100110) [reply tests/test_routes.py:569]: I see, thanks a lot!

**Review by wl2484 — APPROVED** [review]
🎉
